src/main/scala/it/pagopa/interop/apigateway/server/impl/Main.scala

```diff
@@ -0,0 +1,210 @@
+  val vaultService: VaultService = new DefaultVaultService with DefaultVaultClient.DefaultClientInstance
+}
+
+trait AttributeRegistryManagementDependency {
+  val attributeRegistryManagementApi: AttributeApi = AttributeApi(
+    ApplicationConfiguration.attributeRegistryManagementURL
+  )
+
+  val attributeRegistryManagementService =
+    new AttributeRegistryManagementServiceImpl(AttributeRegistryManagementInvoker(), attributeRegistryManagementApi)
+}
+
+trait PurposeManagementDependency {
+  val purposeManagementService = new PurposeManagementServiceImpl(
+    PurposeManagementInvoker(),
+    PurposeApi(ApplicationConfiguration.purposeManagementURL)
+  )
+}
+
+object Main
+    extends App
+    with CORSSupport
+    with VaultServiceDependency
+    with AgreementManagementDependency
+    with AuthorizationManagementDependency
+    with CatalogManagementDependency
+    with PartyManagementDependency
+    with AttributeRegistryManagementDependency
+    with PurposeManagementDependency {
+
+  val dependenciesLoaded: Future[(JWTReader, ClientAssertionValidator, PDNDTokenGenerator)] = for {
+    keyset <- JWTConfiguration.jwtReader.loadKeyset().toFuture
+    jwtReader = new DefaultJWTReader with PublicKeysHolder {
+      var publicKeyset: Map[KID, SerializedKey] = keyset
+      override protected val claimsVerifier: DefaultJWTClaimsVerifier[SecurityContext] =
+        getClaimsVerifier(audience = ApplicationConfiguration.pdndAudience)
+    }
+    clientAssertionValidator = new DefaultClientAssertionValidator with PublicKeysHolder {
+      var publicKeyset: Map[KID, SerializedKey] = keyset
+      override protected val claimsVerifier: DefaultJWTClaimsVerifier[SecurityContext] =
+        getClaimsVerifier(audience = ApplicationConfiguration.pdndAudience)
+    }
+    pdndTokenGenerator = new DefaultPDNDTokenGenerator with PrivateKeysHolder {
+      override val RSAPrivateKeyset: Map[KID, SerializedKey] =
+        vaultService.readBase64EncodedData(ApplicationConfiguration.rsaPrivatePath)
+      override val ECPrivateKeyset: Map[KID, SerializedKey] =
+        vaultService.readBase64EncodedData(ApplicationConfiguration.ecPrivatePath)
+    }
+  } yield (jwtReader, clientAssertionValidator, pdndTokenGenerator)
+
+  dependenciesLoaded.transformWith {
+    case Success((jwtReader, clientAssertionValidator, pdndTokenGenerator)) =>
+      launchApp(jwtReader, clientAssertionValidator, pdndTokenGenerator)
+    case Failure(ex) =>
+      classicActorSystem.log.error(s"Startup error: ${ex.getMessage}")
+      classicActorSystem.log.error(s"${ex.getStackTrace.mkString("\n")}")
+      CoordinatedShutdown(classicActorSystem).run(StartupErrorShutdown)
+  }
+
+  private def launchApp(
+    jwtReader: JWTReader,
+    clientAssertionValidator: ClientAssertionValidator,
+    pdndTokenGenerator: PDNDTokenGenerator
+  ): Future[Http.ServerBinding] = {
+    Kamon.init()
+
+    locally {
+      val _ = AkkaManagement.get(classicActorSystem).start()
```

Comment:
```suggestion
      AkkaManagement.get(classicActorSystem).start()
```
no wartremover warts 😂